tests/Core/Ast/Parser/NikicPhpParser/NikicPhpParserTest.php

```diff
@@ -85,4 +73,67 @@ public function testParseTemplateTypes(): void
         $astClassReferences = $astFileReference->classLikeReferences;
         self::assertCount(0, $astClassReferences[0]->dependencies);
     }
+
+    public function testParseClassDocTags(): void
+    {
+        $parser = $this->getParser();
+        $filePath = __DIR__.'/Fixtures/DocTags.php';
+        $astFileReference = $parser->parseFile($filePath);
+
+        self::assertCount(2, $astFileReference->classLikeReferences);
+        $classesByName = $this->refsByName($astFileReference->classLikeReferences);
+
+        $this->assertSame(
+            [
+                '@internal' => [''],
+                '@note' => ['Note one', 'Note two'],
+            ],
+            $classesByName['TaggedThing']->tags
+        );
+        $this->assertSame([], $classesByName['UntaggedThing']->tags);
+    }
+
+    public function testParseFunctionDocTags(): void
+    {
+        $parser = $this->getParser();
+        $filePath = __DIR__.'/Fixtures/Functions.php';
+        $astFileReference = $parser->parseFile($filePath);
+
+        self::assertCount(2, $astFileReference->functionReferences);
+        $functionsByName = $this->refsByName($astFileReference->functionReferences);
+
+        $this->assertSame(
+            ['@param' => ['string $foo', 'string $bar']],
+            $functionsByName['taggedFunction()']->tags
+        );
+        $this->assertSame([], $functionsByName['untaggedFunction()']->tags);
+    }
+
+    private function refsByName(array $refs): array
+    {
+        $refsByName = [];
+
+        foreach ($refs as $ref) {
+            $name = preg_replace('/^.*\\\\(\w+(\(\))?)$/', '$1', $ref->getToken()->toString());
+            $refsByName[$name] = $ref;
+        }
+
+        return $refsByName;
+    }
+
+    private function getParser(): NikicPhpParser
```

Comment:
It not just getting a value, rename to `buildParser`/`buildSUT`?

Reply:
done